extensions/amp-base-carousel/0.1/amp-base-carousel.js

```diff
@@ -430,6 +476,35 @@ class AmpCarousel extends AMP.BaseElement {
     });
   }
 
+  /**
+   * @return {boolean} Whether or not controls should be hidden.
+   */
+  shouldHideControls_() {
+    if (this.controls_ == Controls.NEVER) {
+      return true;
+    }
+
+    if (this.controls_ == Controls.ALWAYS) {
+      return false;
+    }
+
+    return this.hadTouch_;
+  }
+
+  /**
+   * @param {string} controls
+   * @private
+   */
+  updateControls_(controls) {
+    this.controls_ =
+      controls === 'always'
+        ? Controls.ALWAYS
+        : controls === 'never'
+        ? Controls.NEVER
+        : Controls.AUTO;
```

Comment:
Switch statement for readability? Unless there's some micro-optimization happening here